### PR TITLE
Be consistent with how we report init status.

### PR DESCRIPTION
**Pull request by ncabatoff**
This is the same fix as #9652, together with changing the seal-status api to behave the same as the health and init apis, i.e. use Core.Initialized to govern what we return as our initialized status.

### CODE REVIEW

File: vault/external_tests/raft/raft_test.go [hunk 1]
```diff
@@ -875,3 +875,102 @@ func BenchmarkRaft_SingleNode(b *testing.B) {
...
+	for i := range cluster.Cores {
+		verifyInitStatus(i, i < 1)
+	}
+
+	joinFunc(cluster.Cores[1].Client, false)
+	for i, core := range cluster.Cores {
+		verifyInitStatus(i, i < 2)
+		if i == 1 {
+			cluster.UnsealCore(t, core)
+			time.Sleep(5 * time.Second)
```
Comment (vishalnayak) [inline vault/external_tests/raft/raft_test.go:961]: I am sure you might have thought of not using sleep here. Would it be possible to use `RetryUntil` here?

File: vault/external_tests/raft/raft_test.go [hunk 2]
```diff
@@ -875,3 +875,102 @@ func BenchmarkRaft_SingleNode(b *testing.B) {
...
+		if err != nil {
+			t.Fatal(err)
+		}
+		if health.Initialized != expected {
+			t.Errorf("core %d: expected init=%v, sys/health returned %v", coreIdx, expected, health.Initialized)
+		}
+	}
+
+	for i := range cluster.Cores {
+		verifyInitStatus(i, i < 1)
```
Comment (vishalnayak) [inline vault/external_tests/raft/raft_test.go:953]: The `<` logic is neat!

File: vault/external_tests/raft/raft_test.go [hunk 3]
```diff
@@ -875,3 +875,102 @@ func BenchmarkRaft_SingleNode(b *testing.B) {
...
+			verifyInitStatus(i, true)
+		}
+	}
+
+	joinFunc(cluster.Cores[2].Client, false)
+	for i, core := range cluster.Cores {
+		verifyInitStatus(i, true)
+		if i == 2 {
+			cluster.UnsealCore(t, core)
+			time.Sleep(5 * time.Second)
```
Comment (vishalnayak) [inline vault/external_tests/raft/raft_test.go:971]: Here as well.

**Review by vishalnayak — APPROVED** [review]
Left, optional test comments. Otherwise LGTM!

**Comment by ncabatoff** [comment]
Actually this seems to be incomplete.  Brian's comments in the jira:
>  I think it logically makes sense that a node that has joined a raft cluster but has yet to be unsealed is, for all intents and purposes, initialized

Prior to the unseal/sleep the init status appears as false, which doesn't fit with the above.

**Comment by ncabatoff** [comment]
> Prior to the unseal/sleep the init status appears as false, which doesn't fit with the above.

Correction: it appears as true prior to unseal after the join, as well as after join+unseal, but in between it could report init=false.  This is when we've bootstrapped enough to set Core.raftInfo = nil, but haven't yet written the master key and other things Core.Initialized checks.  Fixed by checking for whether the *raft backend* is initialized in that case, which would happen before we set raftInfo=nil.

**Review by raskchanky — APPROVED** [review]

**Comment by ncabatoff** [comment]
Fixes #9618.

File: vault/external_tests/raft/raft_test.go [hunk 4]
```diff
@@ -875,3 +875,104 @@ func BenchmarkRaft_SingleNode(b *testing.B) {
...
+		verifyInitStatus(i, i < 2)
+		if i == 1 {
+			cluster.UnsealCore(t, core)
+			verifyInitStatus(i, true)
+		}
+	}
+
+	joinFunc(cluster.Cores[2].Client, false)
+	for i, core := range cluster.Cores {
+		verifyInitStatus(i, true)
```
Comment (vishalnayak) [inline vault/external_tests/raft/raft_test.go:967]: Shouldn't this be `i<3`?

Comment (ncabatoff) [reply vault/external_tests/raft/raft_test.go:967]: But i is always less than 3.

File: vault/external_tests/raft/raft_test.go [hunk 5]
```diff
@@ -875,3 +875,104 @@ func BenchmarkRaft_SingleNode(b *testing.B) {
...
+
+	// Seal the leader so we can install an address provider
+	{
+		testhelpers.EnsureCoreSealed(t, leaderCore)
+		leaderCore.UnderlyingRawStorage.(*raft.RaftBackend).SetServerAddressProvider(addressProvider)
+		cluster.UnsealCore(t, leaderCore)
+		vault.TestWaitActive(t, leaderCore.Core)
+	}
+
+	joinFunc := func(client *api.Client, addClientCerts bool) {
```
Comment (vishalnayak) [inline vault/external_tests/raft/raft_test.go:903]: May be the join function can do away with the `addClientCerts` logic as we are only setting it to false.

File: vault/external_tests/raft/raft_test.go (hunk 4 above)
Comment (vishalnayak) [reply vault/external_tests/raft/raft_test.go:967]: Ah, right.

**Review by vishalnayak — APPROVED** [review]

File: vault/external_tests/raft/raft_test.go (hunk 5 above)
Comment (ncabatoff) [reply vault/external_tests/raft/raft_test.go]: Good call.

**Review by raskchanky — APPROVED** [review]

**Pull request merged**